checker/main.go

```diff
@@ -0,0 +1,207 @@
+	"github.com/RedHatInsights/insights-content-service/groups"
+	"github.com/RedHatInsights/insights-results-aggregator/content"
+	"github.com/RedHatInsights/insights-results-aggregator/logger"
+	"github.com/rs/zerolog/log"
+)
+
+// groupConfigMap is a shorthand for the map used to store the group configuration.
+type groupConfigMap map[string]groups.Group
+
+var (
+	groupConfigPath string = "./groups_config.yaml"
+	contentDirPath  string = "./content/"
+)
+
+func init() {
+	flag.StringVar(&groupConfigPath, "config", groupConfigPath, "Path to the group configuration YAML file.")
+	flag.StringVar(&contentDirPath, "content", contentDirPath, "Path to the content directory (the one containing the 'config.yaml' file).")
+	flag.Parse()
+}
+
+func main() {
+	initLogger()
+	groupCfg := checkGroupConfig()
+	checkRuleContent(groupCfg)
+}
+
+// initLogger initializes the zerolog library to pretty-print the log messages.
+func initLogger() {
+	err := logger.InitZerolog(
+		logger.LoggingConfiguration{
+			Debug:                      true,
+			LogLevel:                   "debug",
+			LoggingToCloudWatchEnabled: false,
+		},
+		logger.CloudWatchConfiguration{},
+	)
+	if err != nil {
+		log.Fatal().Err(err).Msg("unable to initialize zerolog")
+	}
+}
+
+// checkGroupConfig reads the group configuration file and performs defined checks on it.
+// Then it returns the config to be used by the rule content checks.
+func checkGroupConfig() groupConfigMap {
+	groupCfg, err := groups.ParseGroupConfigFile(groupConfigPath)
+	if err != nil {
+		log.Fatal().Err(err).Msg("unable to parse group config file")
+	}
+
+	// Check if all tags on all groups are unique.
+	// - If no groups contains the same tag multiple times.
+	// - If no two groups share the same tag name.
+	uniqueTags := map[string]string{}
+	// Unique group is just a check that makes sure no two groups have the same name property.
+	uniqueGroups := map[string]string{}
+
+	// For each group defined in the group configuration file.
+	for groupKey, group := range groupCfg {
+		if firstGroupKey, exists := uniqueGroups[group.Name]; exists {
+			log.Warn().Msgf("multiple groups with the name '%s' (first with key '%s', but also with key '%s')", group.Name, firstGroupKey, groupKey)
+		} else {
+			uniqueGroups[group.Name] = groupKey
+		}
+
+		// For each tag assigned to the group.
+		for _, tag := range group.Tags {
+			if firstGroupName, exists := uniqueTags[tag]; exists {
+				log.Warn().Msgf("tag '%s' is defined multiple times (first time in group '%s', but also in group '%s')", tag, firstGroupName, group.Name)
+			} else {
+				uniqueTags[tag] = group.Name
+			}
+		}
+	}
+
+	return groupCfg
+}
+
+// checkRuleContent checks if rule content files are not empty
+// and if the tags assigned to all error codes really exist.
+func checkRuleContent(groupCfg groupConfigMap) {
+	ruleContentDir, err := content.ParseRuleContentDir(contentDirPath)
+	if err != nil {
+		log.Fatal().Err(err).Msg("unable to parse group config file")
```

Comment:
Nitpick: the error message is not the expected one

Reply:
Good catch, I guess I just copy-pasted that.

Reply:
The new message is still a bit weird :-D You forgot the verb, but as I said, it is a nitpick